### ISSUE TITLE
Build instructions possibly out of date 

### ISSUE THREAD

**Issue by EricYangIBM** [eclipse-openj9/openj9#12712]
In the OpenJDK build instructions "Setting up your build environment without Docker" section (e.g. [v16](https://github.com/eclipse-openj9/openj9/blob/master/doc/build-instructions/Build_Instructions_V16.md#setting-up-your-build-environment-without-docker)) step 2 did not work for me. gcc-9 and g++-9 versions were still used in the build. What I did instead was run the following:  

```
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 1
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 1
```  

[Source](https://askubuntu.com/questions/1229774/how-to-use-an-older-version-of-gcc)

**Comment by pshipton** [comment]
@keithc-ca fyi

**Comment by pshipton** [comment]
So even though gcc 7 was installed, setting `export CC=gcc-7 CXX=g++-7` wasn't enough for the build to use it?

**Comment by EricYangIBM** [comment]
@pshipton No, `gcc --version` showed 9 (same with g++). It gave some C string errors in the build. I also tried `export CC=/usr/bin/gcc` but that didn't work either.

**Comment by pshipton** [comment]
Did you follow the docker instructions to install gcc 7 so that it was available?

**Comment by EricYangIBM** [comment]
Yes. `ls -lh /usr/bin/gcc*` showed that it was installed.

**Comment by keithc-ca** [comment]
`update-alternatives` is probably a better way to select the desired version.

@EricYangIBM What errors did you see building with 9? We should fix those.

**Comment by EricYangIBM** [comment]
```
In file included from /usr/include/string.h:495,
                 from /root/openj9-openjdk-jdk16/openj9/runtime/oti/j9.h:27,
                 from /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:23:
In function 'char* strncpy(char*, const char*, size_t)',
    inlined from 'J9UTF8* getJ9UTF8SignatureFromMethodType(J9VMThread*, j9object_t)' at /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:326:10,
    inlined from '_jobject* Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv*, jclass, jobject, jclass, jint, jboolean)' at /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:702:50:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: error: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp: In function '_jobject* Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv*, jclass, jobject, jclass, jint, jboolean)':
/root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:325:21: note: length computed here
  325 |   UDATA len = strlen(signatures[i]);
      |               ~~~~~~^~~~~~~~~~~~~~~
In file included from /usr/include/string.h:495,
                 from /root/openj9-openjdk-jdk16/openj9/runtime/oti/j9.h:27,
                 from /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:23:
In function 'char* strncpy(char*, const char*, size_t)',
    inlined from 'J9UTF8* getJ9UTF8SignatureFromMethodType(J9VMThread*, j9object_t)' at /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:332:9,
    inlined from '_jobject* Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv*, jclass, jobject, jclass, jint, jboolean)' at /root/openj9-openjdk-jdk16/openj9/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:702:50:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: error: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Configure command: `bash configure --with-freemarker-jar=/root/freemarker.jar --enable-openjdk-methodhandles --with-openj9-cc=/usr/bin/gcc-9 --with-openj9-cxx=/usr/bin/g++-9 --with-boot-jdk=/root/bootjdk15`

**Comment by keithc-ca** [comment]
Those warnings were addressed by #12694.

**Comment by pshipton** [comment]
I think you need to search for `: error`
